### PR TITLE
added 3.11.117 release notes stub and updated latest-tags

### PR DESCRIPTION
**Pull request by sheriff-rh**
Ships 2019-06-26.

@openshift/team-documentation please peer review - need a sanity check for all the lines added!

### CODE REVIEW

File: release_notes/ocp_3_11_release_notes.adoc [hunk 1]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+
+* `CustomResourceDefinition` errors were presented in a confusing manner that made
+troubleshooting difficult. Now, the CRD error messages have been clarified to
+assist in troubleshooting CRD errors.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1702693[*BZ#1702693*])
+
+* There was a missing `@` for an instance variable in the Fluentd remote
+syslog plugin code. In some cases, systemd-journald logged errant values.
+This resulted in rsyslog forwarding failures. Now, the variable has been
+corrected remote logging completes successfully.
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: Missing word here in "Now, the variable has been corrected remote logging completes successfully.". Should there be an "and" between "corrected" and "remote"?

File: release_notes/ocp_3_11_release_notes.adoc [hunk 2]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1703904[*BZ#1703904*])
+
+* Long running Jenkins agent and slave pods would experience defunct
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: Should agent be plural here?

File: release_notes/ocp_3_11_release_notes.adoc [hunk 3]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+process errors, causing a high number of processes to appear in process
+listings until the pod is terminated. Now, `dumb-init` is deployed to clean
+up these defunct processes.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1707448[*BZ#1707448*])
+
+* The environment variable JOURNAL_READ_FROM_HEAD was set to an empty
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: JOURNAL_READ_FROM_HEAD should probably be in backticks here.

File: release_notes/ocp_3_11_release_notes.adoc [hunk 4]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+string.  This caused the default value of `read_from_head` for the journald
+input to be true.  When Fluentd starts up for the first time on a node, it
+reads in the entire journal. This could result in hours of delays for
+system messages to show up in ElasticSearch and Kibana. Now, Fluentd will
+check if the value is set and is not empty, or will use the default value
+of false. Fluentd will read from the tail of the journal when it starts on
+a new node.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1707524[*BZ#1707524*])
+
+* The script 99-origin-dns.sh had a debug flag set to enabled, which would
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: 99-origin-dns.sh should probably be in backticks.

File: release_notes/ocp_3_11_release_notes.adoc [hunk 5]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+
+* Long running Jenkins agent and slave pods would experience defunct
+process errors, causing a high number of processes to appear in process
+listings until the pod is terminated. Now, `dumb-init` is deployed to clean
+up these defunct processes.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1707448[*BZ#1707448*])
+
+* The environment variable JOURNAL_READ_FROM_HEAD was set to an empty
+string.  This caused the default value of `read_from_head` for the journald
+input to be true.  When Fluentd starts up for the first time on a node, it
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: Should true and false in this paragraph (and a few other places in this PR) be in backticks, if they're literal values?

File: release_notes/ocp_3_11_release_notes.adoc [hunk 6]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+a new node.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1707524[*BZ#1707524*])
+
+* The script 99-origin-dns.sh had a debug flag set to enabled, which would
+log debug level messages by default. This has been resolved and debug is
+now set to false.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1707799[*BZ#1707799*])
+
+* Kubernetes pod templates were removed at random. This was because the OpenShift
+Jenkins Sync plugin confused ImagesStreams and ConfigMaps with the sane name
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: Should sane be same?

File: release_notes/ocp_3_11_release_notes.adoc [hunk 7]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+while processing them. An event for one type could delete the pod template
+created for another type. The plugin has been modified to keep track of which
+api object type created the pod template of a given name.
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: api -> API?

File: release_notes/ocp_3_11_release_notes.adoc [hunk 8]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+as meaning that the node hosting the egress IP was not responding.
+High-availability egress IPs may have been switched from one node to another
+spuriously. The monitoring code now distinguishes the case of "egress node not
+responding" from "final destination not responding". High-availability egress
+IPs will not be switched between nodes unnecessarily.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1718542[*BZ#1718542*])
+
+* Refactoring of `openshift_facts` caused the MTU to be improperly set. Hosts
+could not communicate on networks with non-default MTU settings. The
+openshift_facts.py script was updated to properly detect and set the MTU for the
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: openshift_facts.py in backticks.

File: release_notes/ocp_3_11_release_notes.adoc [hunk 9]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+
+* Previously, the web console could show an incorrect "Scaling to..." value for
+stateful sets in the project overview under some conditions. The stateful set
+desired replicas value now correctly updates in the web console project
+overview.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1713211[*BZ#1713211*])
+
+* Previously, a service would not correctly show up in the project overview when
+it selected the `deploymentconfig` label that is automatically set for pods
+created by a deployment config. Now, it correctly show services that select the
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: Again on being consistent with how deploymentconfig/DeploymentConfig/deployment config is used in 3.x.

File: release_notes/ocp_3_11_release_notes.adoc [hunk 10]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1720581[*BZ#1720581*])
+
+[[ocp-3-11-117-enhancements]]
+==== Enhancements
+
+* The Cisco ACI CNI plugin is now available.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1708552[*BZ#1708552*])
+
+* You can now use an Ansible playbook to perform a certificate rotation for the EFK
+stack without needing to run the install/upgrade playbook.This playbook deletes
```
Comment (bergerhoffer) [inline release_notes/ocp_3_11_release_notes.adoc]: Need a space here ".T"

File: release_notes/ocp_3_11_release_notes.adoc [hunk 11]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+This has now been changed to `kube-system`.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1571190[*BZ#1571190*])
+
+* On a director-deployed OpenShift environment, the GlusterFS playbooks
+auto-generate a new heketi secret key for each run. As a result of this,
+operations such as scale out or configuration changes on CNS deployments fail.
+As a workaround, complete the following steps:
+
+. Post-deployment, retrieve the heketi secret key. Use this command on one of the master nodes:
+`sudo oc get secret heketi-storage-admin-secret --namespace glusterfs -o json | jq -r .data.key | base64 -d`
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Yep, I agree. I'll fix it.

File: release_notes/ocp_3_11_release_notes.adoc [hunk 12]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+
+. In an environment file, set the following parameters to that value:
+----
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Done!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 13]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+  openshift_storage_glusterfs_heketi_admin_key
+  openshift_storage_glusterfs_registry_heketi_admin_key
+----
+
+As a result of this workaround, operations such as scale out or configuration
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Done!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 14]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1640382[*BZ#1640382*])
+
+* When a new CA was generated, the certificates on the nodes were not updated and
+would not become ready. Now, the redeploy-certificates playbook will copy the
+certificates and join nodes. Nodes no longer go to a `NotReady` state when
+replacing the CA.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1652746[*BZ#1652746*])
+
+* The oc_adm_router Ansible module allowed edits to add duplicate environment
+variables to the router deploymentconfig. An Ansible inventory file that
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Yep, oversight on my part. Fixed all four instances!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 15]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+variables could produce a deploymentconfig with unpredictable behavior. If an
+edit appends an environment variable to the router deploymentconfig, and a
+variable by that name already exists, the oc_adm_router module now deletes the
+old variable. Using an Ansible inventory file to append environment variables to
+the router deploymentconfig now has predictable behavior and allows users to
+override default environment variable settings.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1656487[*BZ#1656487*])
+
+* A playbook which redeployed master certificates did not update web console
+secrets, causing the web console to fail to start. Now, webconsole secrets are
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 6 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Yes. Whoops! Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 7 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Yep, fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 8 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 9 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed all lower case "deploymentconfig" to "DeploymentConfig".

File: release_notes/ocp_3_11_release_notes.adoc (hunk 10 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 16]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+was not passing a user's token. Now, the token is exchanged to ElasticSearch and
+users with proper roles can retrieve metrics in Prometheus.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1695903[*BZ#1695903*])
+
+* Deploying nodes would fail in the `setup_dns.yaml` playbook during multi-node
+setup. This was resolved by fixing the hostname that was passed to the
+`add_host` function. Now, multi-node setup proceeds as expected.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1698922[*BZ#1698922*])
+
+* Upgrading beteween minor versions would fail because several OpenShift variables
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 17]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+This has now been resolved by a backporting of kube code.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1691023[*BZ#1691023*])
+
+* ElasticSearch could not be monitored with Prometheus because the `oauth-proxy`
+was not passing a user's token. Now, the token is exchanged to ElasticSearch and
+users with proper roles can retrieve metrics in Prometheus.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1695903[*BZ#1695903*])
+
+* Deploying nodes would fail in the `setup_dns.yaml` playbook during multi-node
+setup. This was resolved by fixing the hostname that was passed to the
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 18]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+code to validate host variables. This code has been updated and host variables
+are now stored in the class. As a result, the host variables are not being
+copied on every check. The sanity checks and initialization during upgrades
+takes less time to complete.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1682924[*BZ#1682924*])
+
+* The `oreg_url` variable would not function correctly on disconnected installs
+using Satellite because the etcd image could not perform pulls on disconnected
+installs. Now, guidance and examples have been added to specify the etcd image
+URL isuing `osm_etcd_image` in the associated documentation.
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc [hunk 19]
```diff
@@ -2333,3 +2333,213 @@ To upgrade an existing {product-title} 3.10 or 3.11 cluster to this latest
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1698922[*BZ#1698922*])
+
+* Upgrading beteween minor versions would fail because several OpenShift variables
+were not used during the upgrade process. Now, `api_port` and other
+apiserver-related variables are read during the upgrade process and upgrades
+complete successfully.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1699696[*BZ#1699696*])
+
+* ElasticSearch would fail to start due to invalid certificate dates if hosts had
+non-UTC timezones. When OpenShift Nodes's timezone is not set to UTC, the
```
[the remark this answers is not in the document]
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Yikes, that's a bad one. Thanks! Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 1 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Yep, added. Thanks!

**Pull request merged**